specification/datalake-analytics/data-plane/Microsoft.DataLakeAnalytics/2017-09-01-preview/job.json

```diff
@@ -0,0 +1,1974 @@
+    },
+    "CreateScopeJobParameters": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/CreateJobParameters"
+        }
+      ],
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "the key-value pairs used to add additional metadata to the job information. (Only for use internally with Scope job type.)"
+        }
+      },
+      "description": "The parameters used to submit a new Data Lake Analytics Scope job. (Only for use internally with Scope job type.)"
+    },
+    "CreateJobParameters": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseJobParameters"
+        }
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the friendly name of the job to submit."
+        },
+        "degreeOfParallelism": {
+          "type": "integer",
+          "default": 1,
+          "format": "int32",
+          "description": "the degree of parallelism to use for this job. This must be greater than 0, if set to less than 0 it will default to 1."
+        },
+        "priority": {
+          "type": "integer",
+          "format": "int32",
+          "description": "the priority value to use for the current job. Lower numbers have a higher priority. By default, a job has a priority of 1000. This must be greater than 0."
+        },
+        "logFilePatterns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "the list of log file name patterns to find in the logFolder. '*' is the only matching character allowed. Example format: jobExecution*.log or *mylog*.txt"
+        },
+        "related": {
+          "$ref": "#/definitions/JobRelationshipProperties",
+          "description": "the recurring job relationship information properties."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The parameters used to submit a new Data Lake Analytics job."
+    },
+    "BuildJobParameters": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseJobParameters"
+        }
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the friendly name of the job to build."
+        }
+      },
+      "description": "The parameters used to build a new Data Lake Analytics job."
+    },
+    "BaseJobParameters": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "the job type of the current job (Hive, USql, or Scope (for internal use only)).",
+          "enum": [
+            "USql",
+            "Hive",
+            "Scope"
+          ],
+          "x-ms-enum": {
+            "name": "JobType",
+            "modelAsString": false
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/CreateJobProperties",
```

Comment:
You may want to use `x-ms-flatten` for `CreateJobProperties` type to improve user experience.

Reply:
Yes, this would be ideal.  However, I'm not sure if it's appropriate for this situation because when a user creates a job, we want them to explicit on which properties object to use: CreateScopeJobProperties vs CreateUSqlJobProperties

Reply:
I didn't see that - thanks for explaining!